cont/examples/Widgets/gui_autoshowmetal.lua

```diff
@@ -0,0 +1,44 @@
+function widget:GetInfo()
+  return {
+    name      = "Auto Show Metal",
+    desc      = "Toggles metal view for extractor building commands",
+    license   = "GNU GPL, v2 or later",
+    layer     = -50,
+    enabled   = true
+  }
+end
+
+local isMex = {}
+
+local function initMexes()
+  for uDefID, uDef in pairs(UnitDefs) do
+    if uDef.extractsMetal > 0 then
+      isMex[uDefID] = true
+    end
+  end
+end
+
+function widget:Initialize()
+  if Spring.SetAutoShowMetal then
```

Comment:
Should include forward-compat as well. This is so that games can add the widget before migrating in a fire-and-forget way, and it won't be applying conflicting changes in parallel with the native engine autometal behaviour.
```suggestion
function widget:Initialize()
  if not Engine.FeatureSupport.noAutoShowMetal then
    -- engine hasn't yet removed autometal and handles everything for us! arigato mr. Recoil 🙏
    widgetHandler:RemoveWidget()
    return
  end

  if Spring.SetAutoShowMetal then
```

Reply:
need to check for nil since it's now a bool and want it to not trigger the deprecation warning once this is in, so SetAutoShowMetal needs to be called when it's defined in general.

Reply:
so added a variant of your suggestion.